### DISCUSSION TITLE
Encoding of REST API simplification

### DISCUSSION THREAD

**Discussion by KellerKev** [caprover/caprover#1684] [Q&A] [answered]
I currently deploy my apps using the rest api from caprover. I send the dockerfile and other configuraiton files like a custom nginx config etc. via the res api. I extracted the api calls from the chrome browser. It works but the payload is heavily encoded and make it really hard to maintain these calls or add things like variables to the payload. I currently doing it and it works, buti it is a real pain. For example:

 = '''{\r\n  \"captainDefinitionContent\": \"{\\\"schemaVersion\\\":2,\\\"dockerfileLines\\\":[\\\"FROM

What kind of encoding is this? 

Is there a nicer way of dong it. 

Another example:

payload = '{\"appName\":\"'+iname+'\",\"hasPersistentData\":true}'
            mpayload={}
            mpayload["appname"] = iname
            mpayload["hasPersistentData"] = True    
            t_str = json.dumps(mpayload)
            

Do you know what the difference is between those 2 strings produced here?

**Comment by KellerKev** [comment, marked as answer]
Sorry please disregard. I just did not use json.dumps in python the right way.

**Reply by leomerida15** [reply]
Excuse me, where can I access this API to be able to create new applications in Caprover?
